### PR TITLE
KFSPTS-7351: fix DV copy so vendors aren't cleared incorrectly

### PR DESCRIPTION
**Pull request by bhutchinson**

### CODE REVIEW

**Comment by bhutchinson** [comment]
retest this please

File: src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java [hunk 1]
```diff
@@ -652,6 +653,26 @@ protected void clearFieldsThatShouldNotBeCopied() {
         setTripId(null);
     }
 
+    /**
+     * This overrides the code in the parent base class because of an issue where vendorDetail is null when it shouldn't be.
+     * Might be related to OJB and proxy objects or something like that.
+     * Hopefully we can contribute a fix to base code and eliminate this duplicate method in the future.
+     */
+    protected void clearInvalidPayee() {
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: Should we add an override annotation to this function?

File: src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java [hunk 2]
```diff
@@ -639,7 +641,6 @@ public void toCopy() throws WorkflowException {
 
         super.toCopy();
 
-        KNSGlobalVariables.getMessageList().clear();
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: Just out of curiosity,  was clearing the message list causing part of the problem?

Comment (bhutchinson) [inline src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: It wasn't causing the part where stuff was being cleared that shouldn't, but was causing an issue where messages from the clearing method weren't being displayed like they should be.

File: src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java (hunk 1 above)
Comment (bhutchinson) [inline src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: probably; I can fix that

File: src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java (hunk 2 above)
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: ah, thanks

**Comment by bhutchinson** [comment]
In addition to the one small change, there is one failing test related to this change. It doesn't fail locally, so I'm still trying to figure out what the issue is. We should wait to merge until the tests are green.

File: src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java [hunk 3]
```diff
@@ -91,6 +93,12 @@ public void testToCopy() throws WorkflowException {
         assertNull(cuDisbursementVoucherDocument.getCancelDate());
         assertEquals(KFSConstants.DocumentStatusCodes.INITIATED, cuDisbursementVoucherDocument.getFinancialSystemDocumentHeader().getFinancialDocumentStatusCode());
 
+        MessageList messageList = KNSGlobalVariables.getMessageList();
+        System.err.println("KNSGlobalVariables.getMessageList().size(): " + messageList.size());
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java]: I don't think we want a system.err message here do we?

File: src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java [hunk 4]
```diff
@@ -91,6 +93,12 @@ public void testToCopy() throws WorkflowException {
...
+        for (ErrorMessage errorMessage: messageList) {
+            System.err.println("errorMessage.getErrorKey(): " + errorMessage.getErrorKey());
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java]: another system.err .... shouldn't these be LOG.error?

File: src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java (hunk 3 above)
Comment (bhutchinson) [inline src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java]: It's temporary to help figure out why this test fails in Jenkins but not locally for me.

File: src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java (hunk 4 above)
Comment (bhutchinson) [inline src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentIntegrationTest.java]: again, temporary

**Comment by bhutchinson** [comment]
This should be ready for final review now. The unit test failure is fixed and the temporary debugging is cleaned up.

**Pull request merged**